src/sparseml/pytorch/torchvision/train.py

```diff
@@ -340,45 +339,94 @@ def main(args):
-    )
-
-    collate_fn = None
-    num_classes = len(dataset.classes)
-    mixup_transforms = []
-    if args.mixup_alpha > 0.0:
-        mixup_transforms.append(
-            transforms.RandomMixup(num_classes, p=1.0, alpha=args.mixup_alpha)
+    if args.dataset_path is not None:
+        train_dir = os.path.join(args.dataset_path, "train")
+        val_dir = os.path.join(args.dataset_path, "val")
+        dataset, dataset_test, train_sampler, test_sampler = load_data(
+            train_dir, val_dir, args
         )
-    if args.cutmix_alpha > 0.0:
-        mixup_transforms.append(
-            transforms.RandomCutmix(num_classes, p=1.0, alpha=args.cutmix_alpha)
+        num_classes = len(dataset.classes)
+        collate_fn = None
+
+        mixup_transforms = []
+        if args.mixup_alpha > 0.0:
+            mixup_transforms.append(
+                transforms.RandomMixup(num_classes, p=1.0, alpha=args.mixup_alpha)
+            )
+        if args.cutmix_alpha > 0.0:
+            mixup_transforms.append(
+                transforms.RandomCutmix(num_classes, p=1.0, alpha=args.cutmix_alpha)
+            )
+        if mixup_transforms:
+            mixupcutmix = torchvision.transforms.RandomChoice(mixup_transforms)
+
+            def collate_fn(batch):
+                return mixupcutmix(*default_collate(batch))
+
+        data_loader = torch.utils.data.DataLoader(
+            dataset,
+            batch_size=args.batch_size,
+            sampler=train_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+            collate_fn=collate_fn,
         )
-    if mixup_transforms:
-        mixupcutmix = torchvision.transforms.RandomChoice(mixup_transforms)
-
-        def collate_fn(batch):
-            return mixupcutmix(*default_collate(batch))
-
-    data_loader = torch.utils.data.DataLoader(
-        dataset,
-        batch_size=args.batch_size,
-        sampler=train_sampler,
-        num_workers=args.workers,
-        pin_memory=True,
-        collate_fn=collate_fn,
-    )
 
-    data_loader_test = torch.utils.data.DataLoader(
-        dataset_test,
-        batch_size=args.batch_size,
-        sampler=test_sampler,
-        num_workers=args.workers,
-        pin_memory=True,
-    )
+        data_loader_test = torch.utils.data.DataLoader(
+            dataset_test,
+            batch_size=args.batch_size,
+            sampler=test_sampler,
+            num_workers=args.workers,
+            pin_memory=True,
+        )
+    elif args.use_deeplake:
+        tform = deeplake_transforms.Compose(
+            [
+                deeplake_transforms.RandomRotation(20),  # Image augmentation
+                deeplake_transforms.Resize(args.deeplake_image_size),
+                deeplake_transforms.ToTensor(),  # Must convert to pytorch tensor for subsequent operations to run
+                deeplake_transforms.Normalize([0.5], [0.5]),
+            ]
+        )
+        ds_train = deeplake.load(
```

Comment:
I'd hide it in a helper function in a separate auxiliary script, `train.py` should be fairly compact and readable.